src/app/plugins/checkclasses.plugin.ts

```diff
@@ -0,0 +1,77 @@
+import { GuildChannel } from 'discord.js';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, ClassType } from '../../common/types';
+import Constants from '../../common/constants';
+
+export class CheckClassesPlugin extends Plugin {
+  public name: string = 'Check Class';
+  public description: string = 'lists the classes someone is in';
+  public usage: string = 'checkclass <user>';
+  public pluginAlias = [];
+  public permission: ChannelType = ChannelType.Staff;
+  public pluginChannelName: string = Constants.Channels.Staff.UserOffenses;
+
+  private _MAX_CHAR_LIMIT: number = 2000;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    return args.length >= 1;
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const targetUserName = args.join(' ');
+
+    const member = this.container.guildService
+      .get()
+      .members.filter((m) => m.user.tag === targetUserName)
+      .first();
+    if (!member) {
+      message.reply('User not found.');
+      return;
+    }
+
+    const classes = this.container.classService.getClasses(ClassType.ALL);
+    const chansContainingUser = [];
+
+    for (const classObj of classes) {
+      const [, chan] = classObj;
+      if (chan.permissionOverwrites.get(member.id)?.allow) {
+        chansContainingUser.push(chan);
+      }
+    }
```

Comment:
nit this could be a Array.filter()

Reply:
I don't believe you can filter a Map<string, GuildChannel>.

Reply:
ah right i think its fine then. if we really wanted to, we could take advantage of Map.values() and do something like

`const chansContainingUser = Array.from(classes.values()).filter((class) => Boolean(class.permissionOverwrites.get(member.id)?.allow));`

but this requires 2 passes and we dont wanna upset @c650 😞